class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] l = new int[nums.length];
        int[] r = new int[nums.length];
        l[0] = 1;
        for(int i = 1; i< l.length; i++){
            l[i] = nums[i-1]*l[i-1];
        }
        r[r.length-1] = 1;
        for(int i = r.length-2; i >= 0; i--){
            r[i] = r[i+1] * nums[i+1];
        }
        int[] res = new int[nums.length];
        for(int i = 0; i< l.length; i++){
            res[i] = l[i] * r[i];
        }
        return res;
    }
}  
